ObjectComparator/Comparator/Strategies/Implementations/Collections/DictionaryCompareStrategy.cs

```diff
@@ -36,12 +37,24 @@ public DictionaryCompareStrategy(Comparator comparator) : base(comparator)
 
         public override DeepEqualityResult Compare<T>(T expected, T actual, string propertyName)
         {
-            var genericArguments = expected.GetType().GetGenericArguments();
+            var genericArguments = GetGenericArguments(typeof(T));
             return (DeepEqualityResult) CompareMethod.MakeGenericMethod(genericArguments)
                 .Invoke(this, new[] {(object) expected, actual, propertyName})!;
         }
 
-        public override bool IsValid(Type member) =>
-            member.IsGenericType && member.GetGenericTypeDefinition() == typeof(Dictionary<,>);
+        private static Type[] GetGenericArguments(Type type)
+        {
+            return type.GetInterfaces().Prepend(type)
+                .Where(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IDictionary<,>))
+                .Select(i => i.GetGenericArguments()).First();
+        }
+
+        public override bool IsValid(Type member)
+        {
+            return member.GetInterfaces().Prepend(member)
+                .Where(i => i.IsGenericType)
+                .Select(i => i.GetGenericTypeDefinition())
+                .Contains(typeof(IDictionary<,>));
```

Comment:
please create next field
  private static readonly Type DictType = typeof(Dictionary<,>);
  
  and replace .Contains(DictType);
  
  on line 48 as well